initial base module implementation for Dataplex AutoDQ

thinhha opened the pull request:
REST Resource: projects.locations.dataScans:
https://cloud.google.com/dataplex/docs/reference/rest/v1/projects.locations.dataScans#DataScan

Raw terraform resources:
https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/dataplex_datascan

ludoo commented on modules/dataplex-autodq/variables.tf:
```diff
@@ -0,0 +1,177 @@
+ */
+
+variable "data" {
+  description = "The data source for DataScan. The input variable 'data' should be an object containing a single key-value pair that can be one of: * `entity`: The Dataplex entity that represents the data source (e.g. BigQuery table) for DataScan, of the form: `projects/{project_number}/locations/{locationId}/lakes/{lakeId}/zones/{zoneId}/entities/{entityId}`. * `resource`: The service-qualified  ...
```
variable descriptions are usually one-liners :)

thinhha replied:
haha ok :) will shorten

ludoo commented on modules/dataplex-autodq/variables.tf:
```diff
@@ -0,0 +1,177 @@
+variable "execution_trigger" {
+  description = "The input variable 'execution_trigger' specifies when a scan should be triggered. If not specified, the default is `on_demand`, which means the scan will not run until the user calls `dataScans.run` API. Alternatively, you can set `execution_trigger` to `scheduled` scheduled to run periodically based on a cron schedule expression."
+  type = object({
+    on_demand = optional(object({}))
```
this could also be done differently: the variable is `execution_schedule` with string type, and defaults to `null` which implies on demand.

thinhha replied:
Yes this is a good point. I didn't realize the default behaviour for the official API was on_demand until after I've done everything else. I will refactor this according to your suggestion

ludoo reviewed (approved):
Left a couple minor comments, feel free to resolve them once done (and feel free to ignore the variable name/type change). When you are ready just merge.

Be careful in the future when pushing commits with the wrong email, had to rewrite commit history and force approve the CLA check. :)

ludoo commented on modules/dataplex-autodq/README.md:
```diff
@@ -0,0 +1,273 @@
+# Dataplex AutoDQ 
```
Can you please link the module from the top-level repo README, and the modules folder README?

ludoo replied:
And could you rename the module as `cloud-dataplex-autodq`, or rename the existing `cloud-dataplex` as `dataplex`?

thinhha replied:
The official name is Dataplex not Cloud Dataplex. I will rename the other module accordingly.

thinhha commented:
Thanks Ludo for your feedback. I think I've addressed all of your feedback.

Thank you for also fixing my CLA checks. I'm using the email address I have registered - so not sure why it's failing. Will double check this on Monday.

thinhha commented:
This is getting messy. I'm going to close this PR and reopen with a different branch.
